app/internal/window/os_js.go

```diff
@@ -417,22 +418,17 @@ func (w *window) funcOf(f func(this js.Value, args []js.Value) interface{}) js.F
 func (w *window) animCallback() {
 	w.mu.Lock()
 	anim := w.animating
-	if anim {
-		w.requestAnimationFrame.Invoke(w.redraw)
-	}
 	w.mu.Unlock()
 	if anim {
 		w.draw(false)
 	}
+	w.requestAnimationFrame.Invoke(w.redraw)
 }
 
 func (w *window) SetAnimating(anim bool) {
 	w.mu.Lock()
-	defer w.mu.Unlock()
-	if anim && !w.animating {
-		w.requestAnimationFrame.Invoke(w.redraw)
-	}
 	w.animating = anim
+	w.mu.Unlock()
```

Comment:
Defer the Unlock.

Reply:
It starts inside ` NewWindow`, after the `w.draw(true)`, https://github.com/gioui/gio/pull/7/commits/41d91013e4a6e68c596b630a7c01873427e8e6bf#diff-abf70745e7c00ff72bc8babcc591453b6cde31a320dbb2bc50c13872115855faR85.